Hashnode/umami
# Kieran/personalize analytics page with pub data

## Description

Personalize the advanced analytics page `dashboard -> analytics -> advanced analytics (beta)` with blog details. Previous it just looked like analytics for Hashnode and not blog specific. This includes:

- Meta title, description, ogImage and favicon
- Header logo/title

These should match blogs for the same circumstances.

Vercel deployment of Umami analytics - https://vercel.com/hashnode-prod/hn-analytics-umami

Notes: 

- Since this analytics setup may be temporary (we might move out in the future), it was not worth implementing graphql for the pub data fetching here. GSSP fetch to an api on Main is sufficient. I checked with Bahaa who agreed.

- BEFORE MERGING TO PROD - Set `NEXT_PUBLIC_HASHNODE_ENV` in Vercel

## Testing

- add your `.env` file with the following envs:
  - NEXT_PUBLIC_HASHNODE_ENV=development
  - DATABASE_URL='Find in vercel'
  - HASH_SALT='Find in vercel'
 
- Build with `yarn build`
- Run with `yarn dev -p some other port`
- Run your hashnode servers (using the Hashnode Main `kieran/personalize-analytics-page-with-pub-data` branch).
- Now you can access the advanced analytics `/share` pages locally e.g. `/share/:umamiShareId/:UmamiWebsiteUUID`
- Umami details are on the publication model e.g. `publication.umamiShareId`.
- Test out different publications with different settings
  - Regular pub and Team pub: With/without `logo`, `dark mode logo`, `title`, `seo title`, `seo descrtiption`, `ogImage`, `favicon`.
- Open up the blog in the next tab to help with comparison. Everything should match as it does in Blogs.
